src/session_impl.cpp

```diff
@@ -107,6 +107,8 @@ POSSIBILITY OF SUCH DAMAGE.
 // for logging stat layout
 #include "libtorrent/stat.hpp"
 
+#include <cstdarg> // for va_list
```

Comment:
I think it's reasonable to include what you use directly, and not rely on "collateral" includes by other headers

Reply:
Then it should be
```
#ifndef TORRENT_DISABLE_LOGGING
#include <cstdarg> // for va_list
#endif
```
because of

    #ifndef TORRENT_DISABLE_LOGGING
	TORRENT_FORMAT(2,3)
	void session_impl::session_log(char const* fmt, ...) const
	{
		if (!m_alerts.should_post<log_alert>()) return;

		va_list v;
		va_start(v, fmt);
		session_vlog(fmt, v);
		va_end(v);
	}
    //...
    #endif

Reply:
Good point. Will revise the patch and let you know.

Reply:
Actually for this file the extra include is already put within the guard. I'll revise the other files.